src/libgit2.c

```diff
@@ -48,3 +76,280 @@ int git_libgit2_shutdown(void)
 {
 	return git_runtime_shutdown();
 }
+
+/* Library settings */
```

Comment:
Moving stuff like version or feature information definitely makes a ton of sense to me. But I'm not sure I'm a huge fan of moving settings over, too. I kind of get why you're doing it, but if you ask me we should instead work on clearly separating settings and make them more standalone. Like clearly identifying options with `git_setting_user_agent`, `git_setting_ssl_ciphers` etc.

WDYT?

Reply:
Agree with you here, and I agree that this is continuing to grow in scope.  Let's tackle this in a followup PR?